sycl/cmake/modules/AddSYCLUnitTestPreview.cmake

```diff
@@ -0,0 +1,128 @@
+# add_sycl_unittest(test_dirname SHARED|OBJECT file1.cpp, file2.cpp ...)
+#
+# Will compile the list of files together and link against SYCL.
+# Produces a binary names `basename(test_dirname)`.
+macro(add_sycl_unittest_preview test_dirname link_variant)
```

Comment:
I'd expect preview and non-preview version have a lot of similarities. Is it possible to have

```
define/macro/whatever add_sycl_unittest_impl(bool IsPreview, ...) { ... }
define/macro/whatever add_sycl_unittest(...) {
  add_sycl_unittest_impl(true, ...)
  add_sycl_unittest_impl(false, ...)
}
```

Or just have them both processed in one go in a single routine if that can work (some loop maybe?)

Reply:
Yes, I'm separating out all the common functionality into a different macro.